### PR TITLE
added network selector

### PR DESCRIPTION
**Pull request by arontaubyte**
Added a network selector to either connect to default, deprecated or custom network
Selected network is also tied to the profile. So switching to different profile with a different network will work.

Test: https://asciinema.org/a/6X5d9h2SoEWbmGcTnlzEWRGSy (Updated 06/06)

### CODE REVIEW

**Review by tafseer-khan — CHANGES_REQUESTED** [review]
Also would be nice instead of |grep FAIL 
|grep 'PASS:\|FAIL\|SKIP:'

That way it shows which tests passed, skip, or failed. Someone could put a lot of t.Skip() and a test that just sleeps to fake a recording

File: lib/domain/fqdn.go [hunk 1]
```diff
@@ -62,6 +83,34 @@ func NewGeneratedFQDN(prefix string) (string, error) {
...
+	resp, err := http.Get(fmt.Sprintf("https://seer.tau.%s/network/config", fqdn))
+	if err != nil {
+		return "", fmt.Errorf("fetching generated url prefix for fqdn `%s` failed with: %s", fqdn, err)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("reading response failed with: %s", err)
+	}
+
```
Comment (skelouse) [inline lib/domain/fqdn.go:114]: resp.Body.Close()

File: cli/commands/resources/network/types.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+func (link) Edit() common.Command {
+	return common.NotImplemented
+}
+
+func (link) List() common.Command {
+	return common.NotImplemented
+}
+
+func (link) Delete() common.Command {
+	return nil
```
Comment (skelouse) [inline cli/commands/resources/network/types.go:38]: return common.NotImplemented

File: cli/commands/resources/network/types.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+func (link) New() common.Command {
+	return nil
+}
+
+func (link) Query() common.Command {
+	return nil
```
Comment (skelouse) [inline cli/commands/resources/network/types.go:46]: return common.NotImplemented

File: env/network.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+func GetSelectedNetwork() (string, error) {
+	network, isSet := LookupEnv(constants.CurrentSelectedNetworkName)
+	if isSet == true && len(network) > 0 {
+		return network, nil
+	}
+
+	// Try to get profile from current session
+	network, exist := session.Get().SelectedNetwork()
+	if exist == true && len(network) > 0 {
```
Comment (skelouse) [inline env/network.go:25]: Why check?  returning network, nil either way

File: env/network.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+func GetCustomNetworkUrl() (string, error) {
+	fqdn, isSet := LookupEnv(constants.CustomNetworkUrlName)
+	if isSet == true && len(fqdn) > 0 {
+		return fqdn, nil
+	}
+
+	// Try to get profile from current session
+	fqdn, exist := session.Get().CustomNetworkUrl()
+	if exist == true && len(fqdn) > 0 {
```
Comment (skelouse) [inline env/network.go:48]: same

**Review by skelouse — CHANGES_REQUESTED** [review]
May be better to write it as it's own command, unless adding other resource esqu commands in the future `tau network` would fit.  Placing it in select gives the idea it can be created and destroyed as other resources.  

Also need to add env variables to the README

File: singletons/auth_client/client.go [hunk 6]
```diff
@@ -20,12 +22,22 @@ func Clear() {
 }
 
 func getClientUrl() (url string) {
-	url = os.Getenv(constants.AuthURLEnvVarName)
-	if url == "" {
+	selectedNetwork, _ := env.GetSelectedNetwork()
+	switch selectedNetwork {
+	case common.DefaultNetwork:
```
[the remark this answers is not in the document]
Comment (arontaubyte) [reply singletons/auth_client/client.go:28]: added as todo.

**Review by tafseer-khan — APPROVED** [review]

**Pull request merged**